src/Controls/src/Core/HandlerImpl/RoundRectangle.Impl.cs

```diff
@@ -32,5 +33,37 @@ public override PathF GetPath()
 
 			return path;
 		}
+
+		PathF IRoundRectangle.InnerPathForBounds(Rect viewBounds, float strokeWidth)
+		{
+			if (HeightRequest < 0 && WidthRequest < 0)
+			{
+				Frame = viewBounds;
+			}
+
+			var path = GetInnerPath(strokeWidth);
+
+			return path;
+		}
+
+		internal PathF GetInnerPath(float strokeWidth)
```

Comment:
We was clipping using the Border shape. That is correct in many cases but not always. Using a RoundRectangle, the Corner Radius value is not the same in the inner or outer. We want in this case user the inner value. For that reason, added a new method returning the same shape but with different radius.

![](https://i.stack.imgur.com/XWsyG.png)